### PR TITLE
[13.0][IMP] website_sale_product_minimal_price: Forward port of improvements of 12.0

### PR DESCRIPTION
**Pull request by CarlosRoca13**
Forward port adapting the minimal price computing to the new way added on version 13.0.

With the changes added the price takes care on the website selected pricelist and it's added the grid of price scale for each product.variant.

cc @Tecnativa TT24175

please @sergio-teruel @pedrobaeza review this.

### CODE REVIEW

File: website_sale_product_minimal_price/__manifest__.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
     "license": "AGPL-3",
     "application": False,
     "installable": True,
-    "depends": ["website_sale"],
+    "depends": ["website_sale_stock"],
```
Comment (pedrobaeza) [inline website_sale_product_minimal_price/__manifest__.py]: Why changing this dependency?

File: website_sale_product_minimal_price/controllers/main.py [hunk 2]
```diff
@@ -19,30 +19,74 @@ def get_combination_info_minimal_price(self, product_template_ids, **kw):
...
+                    ("product_tmpl_id", "=", product.product_tmpl_id.id),
+                    (
+                        "categ_id",
+                        "in",
+                        list(map(int, product.categ_id.parent_path.split("/")[0:-1])),
+                    ),
+                    ("min_quantity", ">", 0),
+                ]
+            )
+            .mapped("min_quantity")
```
Comment (pedrobaeza) [inline website_sale_product_minimal_price/controllers/main.py]: Do this in 2 steps for better black formatting: first obtain items records, and then apply `mapped`.

File: website_sale_product_minimal_price/__manifest__.py (hunk 1 above)
Comment (CarlosRoca13) [reply website_sale_product_minimal_price/__manifest__.py]: Because the module that adds the method `VariantMixin._onChangeCombinationStock` is this other

Comment (pedrobaeza) [reply website_sale_product_minimal_price/__manifest__.py]: And can we avoid that one and use one in pure `website_sale` module? It's not logical to require stock for this. What about if you sell services?

**Comment by CarlosRoca13** [comment]
@pedrobaeza change done

File: website_sale_product_minimal_price/controllers/main.py [hunk 3]
```diff
@@ -19,30 +19,70 @@ def get_combination_info_minimal_price(self, product_template_ids, **kw):
...
+        """Special route to use website logic in get_combination_info override.
+        This route is called in JS by appending _website to the base route.
+        """
+        product = request.env["product.product"].browse(product_id)
+        pricelist = request.env["website"].get_current_website().get_current_pricelist()
+        # Getting all min_quantity of the current product to compute the possible
+        # price scale.
+        qty_list = request.env["product.pricelist.item"].search(
+            [
+                ("|"),
```
Comment (pedrobaeza) [inline website_sale_product_minimal_price/controllers/main.py]: I'm seeing that this is incorrect syntax. It should be just:
```suggestion
                "|",
```

If it works, it's by chance.

File: website_sale_product_minimal_price/__manifest__.py (hunk 1 above)
Comment (CarlosRoca13) [reply website_sale_product_minimal_price/__manifest__.py]: Ok I will try to use the website_sale module

**Comment by CarlosRoca13** [comment]
All changes done @pedrobaeza

**Review by pedrobaeza — APPROVED** [review]

**Review by chienandalu — COMMENTED** [review]
Can you point to the fowarded commits?

**Comment by CarlosRoca13** [comment]
@chienandalu yes sure, here you have them:
- https://github.com/OCA/e-commerce/commit/35f35142060413965523bd75f757798fecbbd368#diff-d3f954ac4904966e21fd42f7b88212a43b229ad5c025afc848408a4127215ebe
- https://github.com/OCA/e-commerce/commit/bac82bf23b59241d097ae37b59873bbb6fe5e3f3#diff-d3f954ac4904966e21fd42f7b88212a43b229ad5c025afc848408a4127215ebe
- https://github.com/OCA/e-commerce/commit/d81b0075f1f242b4bb267e0bc21a8e6b5f1b7ce4#diff-d3f954ac4904966e21fd42f7b88212a43b229ad5c025afc848408a4127215ebe
- https://github.com/OCA/e-commerce/commit/3964915256789e326902dd4fbcc40eb08bfade71#diff-d3f954ac4904966e21fd42f7b88212a43b229ad5c025afc848408a4127215ebe
- https://github.com/OCA/e-commerce/commit/ac744844b441cb6ac6fd9e6ce3fd82e142359f3f#diff-d3f954ac4904966e21fd42f7b88212a43b229ad5c025afc848408a4127215ebe

And It's added the new grid price compute from this PR: https://github.com/OCA/e-commerce/pull/539

**Comment by CarlosRoca13** [comment]
Added the changes accepted on this PR https://github.com/OCA/e-commerce/pull/539 doing the necessary changes for allow working with the new version

**Pull request merged**